snuba/web/query.py

```diff
@@ -47,6 +56,47 @@ def __init__(
         self.meta = meta
 
 
+@dataclass(frozen=True)
+class ClickhouseQueryMetadata:
+    sql: str
+    timing: Mapping[str, Any]
+    stats: Mapping[str, Any]
+    status: str
+    trace_id: str
+
+    def to_dict(self):
+        return {
+            "sql": self.sql,
+            "timing": self.timing,
+            "stats": self.stats,
+            "status": self.status,
+            "trace_id": self.trace_id,
+        }
+
+
+@dataclass
+class SnubaQueryMetadata:
+    request: Request
+    dataset: Dataset
+    timer: Timer
+    query_list: MutableSequence[ClickhouseQueryMetadata]
+    http_referrer: Optional[str] = ""
```

Comment:
There's a longer comment below about why, but we shouldn't/can't use HTTP referer here. `request.referer` already has it, anyway.

Reply:
Ok, I copied it from https://github.com/getsentry/snuba/blob/master/snuba/web/query.py#L280, is there some reason we need to be using referrer from http_request there?

Reply:
I don't think so, I can't think of any reason why that can't/shouldn't also use the non-Flask request object. I think the only reason this isn't erroring during subscriptions is because we aren't using the APM APIs. It doesn't help that these functions are in `snuba.web` (since up until recently they were very coupled to the web request/response cycle) but also used for general query execution, so it'd be totally reasonable to assume that relying on the web request is fine here.